File: tests/es_bots/es_refsTest.py
from __future__ import annotations

from pathlib import Path

from src.es_bots.es_refs import mv_es_refs


FIXTURES_DIR = Path(__file__).parent / "texts"


class es_refsTest:
    def testFileText(self) -> None:
        text_input = (FIXTURES_DIR / "es_refs_input.txt").read_text(encoding="utf-8")
        text_output = (FIXTURES_DIR / "es_refs_output.txt").read_text(encoding="utf-8")
        result = mv_es_refs(text_input)
        (FIXTURES_DIR / "es_refs_output_fixed.txt").write_text(result, encoding="utf-8")
        assert text_output.replace("\r\n", "\n") == result.replace("\r\n", "\n")

File: tests/es_bots/es_refsTest.php
<?php

use FixRefs\Tests\MyFunctionTest;
use function WpRefs\EsBots\es_refs\mv_es_refs;

class es_refsTest extends MyFunctionTest
{
    public function testFileText()
    {
        $text_input   = file_get_contents(__DIR__ . "/texts/1/input.txt");
        $text_output  = file_get_contents(__DIR__ . "/texts/1/expected.txt");
        $file_3  = __DIR__ . "/texts/1/output.txt";
        // --
        $result = mv_es_refs($text_input);
        // --
        $result = preg_replace("/\r\n/", "\n", $result);
        $text_output = preg_replace("/\r\n/", "\n", $text_output);
        // --
        file_put_contents($file_3, $result);
        // --
        $this->assertEquals($text_output, $result, "Unexpected result");
    }
}
